src/plugins/inspector/common/adapters/request/request_adapter.ts

```diff
@@ -68,6 +69,27 @@ export class RequestAdapter extends EventEmitter {
     return Array.from(this.requests.values());
   }
 
+  /**
+   * Extract the warning strings from Elasticsearch search responses
+   */
+  public extractWarnings(): ResponseWarning[] | undefined {
+    const response = Array.from(this.requests.values())
+      .filter((req) => {
```

Comment:
`req.response.json` should be `estypes.SearchResponse` here?  there doesn't seem to be a `.warning` on it ?

maybe we are looking for `(req.response.json as estypes.SearchResponse<unknown>)._shards.failed` ?

Reply:
I'm not sure what's the best way to tackle types issues here, since Inspector "is not coupled to a specific implementation or even Elasticsearch". It looks like this is why `json` is typed as `object` instead of anything to do with estypes.

Extracting warnings this way carries on the way warnings were grabbed from the response in the previous implementation. Maybe you can suggest something if you have it in mind.

Reply:
The `response` object is `IEsSearchResponse<any>`. 

https://github.com/elastic/kibana/blob/8.4/src/plugins/data/common/search/types.ts#L73-L76

I'm not sure if we should tie things up from the inspector side, or leave it less coupled.

Reply:
^^ https://github.com/elastic/kibana/pull/138342/files#r944190252

Reply:
resolved - no more changes to inspector RequestAdapter in this PR